library/rendering/entity_rendering/src/main/java/org/quiltmc/qsl/rendering/entity/impl/client/ArmorProviderPhaseData.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2021-2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.rendering.entity.impl.client;
+
+import java.lang.reflect.Array;
+import java.util.Arrays;
+
+import org.jetbrains.annotations.NotNull;
+
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemConvertible;
+import net.minecraft.util.Identifier;
+
+import org.quiltmc.qsl.base.api.phase.PhaseData;
+
+public final class ArmorProviderPhaseData<T> extends PhaseData<ArmorProviderPhaseData.Provider<T>[], ArmorProviderPhaseData<T>> {
```

Comment:
I admit not having expected this ahah.

I thought it would have been internal events.
We might want to switch to that actually since Events are thread-safe and this code doesn't seem to be thread-safe?
And it shouldn't be too bad to translate, unless there's a good reason to have our own handling.

Reply:
Oh right, I left the `lock` out when copying Event's code
I couldn't think of a better way to store the applicable items.

Reply:
Would storing an Event instance on the items work?

Reply:
No, because `addPhaseOrdering` is independent from the applicable items.

Reply:
but is that an issue?
since the idea of the event is you add the listener to all items separately, same for the ordering.